src/components/organisms/UiMenu/stories/HasButton.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+  <UiPopover class="has-button">
+    <UiMenu
+      v-bind="args"
+      role="listbox"
+      :aria-label="ariaLabel"
+      aria-multiselectable="true"
+      :aria-activedescendant="ariaActivedescendant"
+      :aria-busy="ariaBusy"
+    >
+      <template #[`menu-item-${index}`] v-for="(item, index) in itemsToRender">
+        <template v-if="item.hasCustomOption">
+          <UiText
+            tag="span"
+            class="ui-menu-item-button__title">
+            <UiIcon
+              :icon="item.icon"
+              class="ui-button__icon" />
+            <UiText
+              tag="span"
+              class="ui-menu-item-button__label"
+              v-bind="item.labelButtonAttrs">
+              {{ item.label }}
+            </UiText>
+          </UiText>
+          <UiText
+            tag="span"
+            class="ui-menu-item-button__hint">
+            {{ item.hint }}
+          </UiText>
+        </template>
+        <template v-else>
+          <UiCheckbox :aria-label="item.label" disabled>{{ item.label }}</UiCheckbox>
+        </template>
+      </template>
+    </UiMenu>
+  </UiPopover>
+</template>
+
+<script setup lang="ts">
+import {
+  computed,
+  useAttrs,
+  defineOptions,
+} from 'vue';
+import {
+  UiButton,
+  UiCheckbox,
+  UiIcon,
+  UiMenu,
+  UiPopover,
+  UiText,
+} from '@infermedica/component-library';
+
+defineOptions({ inheritAttrs: false });
+const attrs = useAttrs();
+const args = computed(() => (attrs));
+
+export interface AddEmits {
+  (e:'add'): void;
+}
+
+const emit = defineEmits<AddEmits>();
+const clickHandler = () => {
+  emit('add');
+};
+
+const itemsToRender = computed(() => {
+  let items = [attrs.items, {
+    label: 'Didn\'t find chronic condition?',
+    hint: 'Add with your own words',
+    tag: UiButton,
+    icon: 'plus',
+    role: 'option',
+    hasCustomOption: true,
+  }].flat();
+
+  items = items.map((item, index) => {
+    if (typeof item === 'object') {
+      return {
+        ...item,
+        "aria-selected": false,
+        "aria-setsize": items.length,
+        "aria-posinet": index + 1,
+      };
+    }
+  });
+
+  return items.slice(0, attrs.hasButton ? items.length : items.length - 1);
+});
```

Comment:
ok, let we clean this part of code. 
what we want to do? 

> * Is always displayed as the last Option in Listbox.
>   * So, it should not be displayed until all Pre-defined Options (or all matching the > user’squery) were loaded into Listbox.

```
const allPredefinedOptionsAreLoaded = ref(false);

onMounted(()=>{
setTimeout(()=>{
allPredefinedOptionsAreLoaded.value = true;
}, 1000)
})

const itemsToRender = computed(()=>{
const customOption = {
  name: 'custom-option'
}
return [
...items, 
allPredefinedOptionsAreLoaded 
  ? customOption 
  : {}
].map((item, index)=>({
...item,
"aria-selected": false,
"aria-setsize": items.length,
"aria-posinet": index + 1,
}))
})
```
and we should make a changes in template because when we add customOption with name `custom-option` we will create slot with name `custom-option` because in UiMenu it's looks like that 
```
 <slot
            v-bind="item"
            :name="item.name" // menu-item... it's a fallback for items without name, but you we will name custom-option :) 
          >
            <UiText tag="span">{{ item.label }}</UiText>
          </slot>
```
then you should update `<template/>`
```
...
<template #[`menu-item-${index}`] v-for="(item, index) in itemsToRender">
  <UiCheckbox :aria-label="item.label" disabled>{{ item.label }}</UiCheckbox>
</template>
<template #custom-option>
<!-- static template for Custom Option -->
</template>

```